### PR TITLE
fix: fix seed with multiple ranks

### LINKED ISSUE #4440: [BUG] seed is unsafe in TF parallel training
### Bug summary

Per https://numpy.org/doc/stable/reference/random/parallel.html#sequence-of-integer-seeds

> For example, it is common to see users add the worker ID to the root seed, especially with the legacy [RandomState](https://numpy.org/doc/stable/reference/random/legacy.html#numpy.random.RandomState) code.
>
> ```py
> # UNSAFE! Do not do this!
> worker_seed = root_seed + worker_id
> rng = np.random.RandomState(worker_seed)
> ```
>
> It is true that for any one run of a parallel program constructed this way, each worker will have distinct streams. However, it is quite likely that multiple invocations of the program with different seeds will get overlapping sets of worker seeds. It is not uncommon (in the author’s self-experience) to change the root seed merely by an increment or two when doing these repeat runs. If the worker seeds are also derived by small increments of the worker ID, then subsets of the workers will return identical results, causing a bias in the overall ensemble of results.

Unlucky, our TF codes use such the logic, as found in https://github.com/deepmodeling/deepmd-kit/pull/4435#discussion_r1861425911

https://github.com/deepmodeling/deepmd-kit/blob/e695a91ca6f7a1c9c830ab1c58b7b7a05db3da23/deepmd/tf/entrypoints/train.py#L220-L225

### DeePMD-kit Version

devel

### Backend and its version

-

### How did you download the software?

Built from source

### Input Files, Running Commands, Error Log, etc.

See above

### Steps to Reproduce

See above

### Further Information, Files, and Links

_No response_

### PR DESCRIPTION
**Pull request by njzjz**
Fix #4440.

## Summary by CodeRabbit

- **New Features**
	- Enhanced seed handling to support both single integers and lists for improved randomness in distributed training.
	- Added logging for neighbor statistics calculation during training.

- **Bug Fixes**
	- Improved error handling in data loading processes to ensure robustness.

- **Documentation**
	- Updated documentation for methods related to seed and batch size management for clarity.

### CODE REVIEW

File: deepmd/pt/utils/dataloader.py [hunk 1]
```diff
@@ -50,8 +50,12 @@
...
 
 def setup_seed(seed) -> None:
-    torch.manual_seed(seed)
-    torch.cuda.manual_seed_all(seed)
+    if isinstance(seed, (list, tuple)):
+        mixed_seed = mix_entropy(seed)
+    else:
+        mixed_seed = seed
+    torch.manual_seed(mixed_seed)
+    torch.cuda.manual_seed_all(mixed_seed)
```
[the remark this answers is not in the document]
Comment (njzjz) [reply deepmd/pt/utils/dataloader.py:59]: Fixed

**Review by wanghan-iapcm — APPROVED** [review]

**Review by caic99 — APPROVED** [review]
